docs/bits/type/function.md

```diff
@@ -0,0 +1,25 @@
+**Function Domains**
+
+Takes 2 ```domains```(this page yet to be written) as arguments: the ```defined``` set and the ```range``` set of the function.
+The ```defined```set is the domain of the function - the set of inputs for the function
+the ```range``` set is the codomain of the function - the set of possible outputs for the function
+
+```essence
+**function** < comma separated attributes(optional)> <defined> --> <range>
+```
+There are 3 groups of function attributes
+  ~ related to cardinality = ```size```, ```minSize```, and ```maxSize```. They take arguments
+  ~ related to function properties = ```injective```, ```surjective```, ```bijective```
+  ~ related to partiality = ```total```
+
+  Functions are partial by default
+
+  You can also explicity define a function as seen [here.](https://github/conjure-cp/conjure/docs/notebooks/functionDemonstration.ipynb)
+
+  See demonstrations of the function attributes [here.](https://github/conjure-cp/conjure/docs/notebooks/functionDemonstration.ipynb)
+
+  Or see functions used to solve the Magic Hexagon Problem [here.](https://github.com/conjure-cp/conjure/tree/main/docs/notebooks/magicHexagon.ipynb)
+
+  Much of the informatin about functions lifted from [the readthedocs.](https://conjure.readthedocs.io/en/latest/essence.html)
```

Comment:
`informatin`: should we run a spellchecker on the whole of `docs/` maybe?

Reply:
I think that is a good idea.